src/cmd_line/subparsers/file.ts

```diff
@@ -11,12 +11,21 @@ export function parseEditFileCommandArgs(args: string): node.FileCommand {
     var scanner = new Scanner(args);
     scanner.skipWhiteSpace();
 
-    if (scanner.isAtEof) {
-        return new node.FileCommand({});
+    let name = "";
+    while (true) {
+        if (scanner.isAtEof) {
+            break;
+        }
+        let c = scanner.next();
+
+        if (/\s/.test(c)) {
+            break;
+        } else {
+            name += c;
+        }
```

Comment:
It actually seems to me like scanner itself should have a method called `readNextWord`, which this should go into.